src/cdk-experimental/menu/menu-item.ts

```diff
@@ -106,8 +114,11 @@ export class CdkMenuItem implements FocusableOption {
 
   /** Get the label for this element which is required by the FocusableOption interface. */
   getLabel(): string {
-    // TODO(andy): implement a more robust algorithm for determining nested text
-    return this._elementRef.nativeElement.textContent || '';
+    // we know that the current node is an element type
+    const clone = this._elementRef.nativeElement.cloneNode(true) as Element;
```

Comment:
I can imagine cloning the entire subtree becoming expensive if we have to do it often. It might be better to traverse the DOM and concatenate the string instead.

Reply:
I brought this up with @jelbourn and he mentioned that it should be ok for now.

Reply:
Yeah, I figure it's not worth worry about unless we find that the implementation ends up being slow in practice